### ISSUE TITLE
Make the main loop of net_dynamic_hb more readable

### ISSUE THREAD

**Issue by vkomenda** [poanetwork/hbbft#373]
The main loop of `net_dynamic_hb` apparently has too many internal loops. Clippy throws an error `cyclomatic_complexity` if not is silenced. The loop has to be split into several functions accessing shared [test state](https://github.com/poanetwork/hbbft/blob/742ad7b83a3f613fd9a7b4ff88cfa82b10bd7dd3/tests/net_dynamic_hb.rs#L452).

**Comment by trendsetter37** [comment]
Is this issue outdated? It seems as though `net_dynamic_hd` has been updated/removed.

**Comment by afck** [comment]
It has been replaced/renamed, but it's still [spaghetti code](https://github.com/poanetwork/hbbft/blob/059e4471cb3581489473ff29b93c233373ddf752/tests/dynamic_honey_badger.rs#L89).

**Comment by trendsetter37** [comment]
I'll take a wack at this.

**Comment by trendsetter37** [comment]
@afck is `do_drop_and_re_add` [function(https://github.com/poanetwork/hbbft/blob/master/tests/dynamic_honey_badger.rs#L89) still invoked when running `cargo test`? I didn't get any errors/warnings when removing the `clippy_cyclomatic_complexity` flag. Also is there an IRC or slack channel associated with this project?

**Comment by afck** [comment]
Yes, that's a clippy lint. Also, `cargo clippy` doesn't check the test code. You need to use `cargo clippy --tests` or `cargo clippy --all-targets` for that.
